client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/main/MainScreenModel.kt

```diff
@@ -35,13 +35,21 @@ class MainScreenModel(
         updateState {
             it.copy(
                 username = username,
-                firstUsernameLetter = username.first().uppercase()
+                firstUsernameLetter = username.first().uppercase(),
+                hasInternetConnection = true
             )
         }
     }
 
     private fun onError(error: ErrorState) {
-        updateState { it.copy(error = error) }
+        when(error){
+            is ErrorState.NoConnection -> {
+                updateState { it.copy(error = error, hasInternetConnection = true) }
```

Comment:
u can remove `error ` i think we don't need it

Reply:
if not specified will not mapp the right error, I made it seperate.